src/lib/components/BaseResponseElement/index.js

```diff
@@ -0,0 +1,256 @@
+        this.enableOption(option);
+      } else if (!isSelected && !hasBeenChecked) {
+        this.disableOption(option);
+      }
+    }
+  }
+
+  // Return true if selections include all correct options.
+  // NOTE: This considers responses that include all correct options
+  // AND some incorrect options as correct--because it's not clear what else
+  // we'd do in that case other than reveal the full correct answer.
+  checkIfCorrect() {
+    const correctAnswersArr = this.correctAnswer.split(",").map(Number);
+    const options = this.querySelectorAll("[data-role=option]");
+    const selectedOptions = this.querySelectorAll(
+      "[data-role=option][data-selected]",
+    );
+    const selections = [];
+
+    for (let i = 0; i < selectedOptions.length; i++) {
+      const index = Array.from(options).indexOf(selectedOptions[i]);
+
+      selections.push(index);
+    }
+    return correctAnswersArr.every((val) => selections.includes(val));
+  }
+
+  // Updates option states when response component is submitted.
+  // NOTE: Assumes client components have a deselectOption() method.
+  // (Necessary because selection mechanisms will vary by response type.)
+  submitResponse() {
+    const options = this.querySelectorAll("[data-role=option]");
+
+    for (const option of options) {
+      const isSelected = option.hasAttribute("data-selected");
+      const isCorrect = option.hasAttribute("data-correct");
+      const isSubmitted = option.hasAttribute("data-submitted");
+
+      if (this.state === "answeredIncorrectly") {
+        if (isSelected && isCorrect) {
+          option.setAttribute("data-submitted", "");
+          this.disableOption(option);
+        } else if (isSelected && !isCorrect) {
+          option.setAttribute("data-submitted", "");
+          this.disableOption(option);
+          this.deselectOption(option);
+        } else if (!isSelected && !isSubmitted) {
+          this.enableOption(option);
+        }
+      } else if (this.state === "answeredCorrectly") {
+        this.disableOption(option);
+        if (isSelected) {
+          option.setAttribute("data-submitted", "");
+        }
+      }
+    }
+    // Force responseComponent state change on next option selection
+    // so AssessmentQuestion state updates.
+    if (this.state === "answeredIncorrectly") {
+      this.state = "unanswered";
+    }
+  }
+
+  // Helper function to allow other components to reset the response
+  // to its unanswered state.
+  // NOTE: Assumes client components have a deselectOption() method.
+  // (Necessary because selection mechanisms will vary by response type.)
+  reset() {
+    const options = this.querySelectorAll("[data-role=option]");
+
+    this.state = "unanswered";
+    for (const option of options) {
+      option.removeAttribute("data-submitted");
+      if (typeof this.deselectOption == "function") {
+        this.deselectOption(option);
+      }
+      this.enableOption(option);
+    }
+  }
+
+  disableOption(option) {
```

Comment:
If you disable the fieldset contained within this element, it'll disable all contained form elements automagically.

https://twitter.com/samthor/status/1147081661168799744

Reply:
Oh, nice, but we don't want _all_ the form elements in the fieldset to disable--just particular ones depending on context. (For example, when the user selects the max number of options allowed, the remaining options disable. But all options are in the same fieldset because it's a single radio group.)